src/Migration.php

```diff
@@ -181,7 +197,38 @@ public function setModel(Model $m)
                 continue;
             }
 
-            $this->field($field->actual ?: $field->short_name, ['type' => $field->type]);  // todo add more options here
+            // get field type from field
+            $type = $field->type;
+
+            // if the field is a hasOne relation
+            // Don't have the right FieldType
+            // FieldType is stored in the reference field
+            if ($field->reference instanceof HasOne) {
+
+                // @TODO if this can be done better?
+
+                // i don't want to :
+                // - change the isolation of relation link
+                // - expose the protected property ->their_field
+                // i need the type of the field to be used in this table
+                $reflection = new \ReflectionClass($field->reference);
+                $property = $reflection->getProperty('their_field');
+                $property->setAccessible(true);
+
+                // get their_field id
+                $reference_their_field = $property->getValue($field->reference);
+                $their_field = $reference_their_field ?? $field->reference->owner->id_field;
+
+                $refModelClass = $field->reference->model;
+
+                /** @var Model $refModel */
+                $refModel = new $refModelClass($m->persistence);
+                $refModel->getField($their_field);
+
+                $type = $refModel->getField($their_field)->type ?? 'integer';
```

Comment:
what about also checking $m->getField($refModel->our_field ?: 'id')->type before falling back to integer?

Reply:
i do it before, i should change the name the var to `$reference_field` in place of `$their_filed` which is misleading

on line 220 : `$their_field = $reference_their_field ?? $field->reference->owner->id_field;`

Reply:
i changed comment and naming to be less missleading